client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java

```diff
@@ -44,6 +44,61 @@
 
 public class SortBasedPusher extends MemoryConsumer {
 
+  class MemoryThresholdManager {
+
+    private final long maxMemoryThresholdInBytes;
+    private final double smallPushTolerateFactor;
+
+    MemoryThresholdManager(
+        int numPartitions, long sendBufferSizeInBytes, double smallPushTolerateFactor) {
+      this.maxMemoryThresholdInBytes = numPartitions * sendBufferSizeInBytes;
```

Comment:
Hi @CodingCat , after rethinking, I think we still need a configuration to limit the max memory threshold instead of calculating by `numPartitions * sendBufferSizeInBytes`. Say we have 100,000 partition, and send buffer uses default 64K, then threshold would be 6.4G, it's too large and not what we wanted.